pip/pip-342 OTel client metrics support.md

```diff
@@ -0,0 +1,167 @@
+    ConsumerStats getStats();
+}
+```
+
+## Initial set of metrics to include
+
+Based on the experience of Pulsar Go client SDK metrics (
+see: https://github.com/apache/pulsar-client-go/blob/master/pulsar/internal/metrics.go),
+this is the proposed initial set of metrics to export.
+
+Additional metrics could be added later on, though it's better to start with the set of most important metrics
+and then evaluate any missing information.
+
+These metrics names and attributes will be considered "Experimental" for 3.3 release and might be subject to changes.
+The plan is to finalize all the namings in 4.0 LTS release.
+
+Attributes with `[name]` brackets will not be included by default, to avoid high cardinality metrics.
+
+| OTel metric name                                | Type          | Unit        | Attributes                                                                                                                                             | Description                                                                                                                               |
+|-------------------------------------------------|---------------|-------------|--------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------|
+| `pulsar.client.connection.opened`               | Counter       | connections |                                                                                                                                                        | The number of connections opened                                                                                                          |
+| `pulsar.client.connection.closed`               | Counter       | connections |                                                                                                                                                        | The number of connections closed                                                                                                          |
+| `pulsar.client.connection.failed`               | Counter       | connections |                                                                                                                                                        | The number of failed connection attempts                                                                                                  |
+| `pulsar.client.producer.opened`                 | Counter       | sessions    | `pulsar.tenant`, `pulsar.namespace`, [`pulsar.topic`], [`pulsar.partition`]                                                                            | The number of producer sessions opened                                                                                                    |
```

Comment:
`pulsar.producer.name/id` should be added.

Reply:
Consumer metrics are also the same.

Reply:
I feel this would be a bit too much information attached. We can revisit later on.

Reply:
Ok, just a suggestion.